lib/auth.ts

```diff
@@ -120,6 +131,37 @@ export class Auth {
     return token;
   };
 
+  /**
+   * Refresh auth token silently for browser environments
+   */
+  public refreshTokenSilently = async (): Promise<ClientOAuth2.Token> => {
+    if (!document || !window) {
+      throw new SilentAuthorizationError({
+        message:
+          "Silent auth token refresh is only available in browser environments"
+      });
+    }
+
+    const iframeUri = await this.getAuthUri({
+      query: {
+        prompt: "none",
+        response_mode: "web_message"
+      }
+    });
+
+    try {
+      const code = await authorizeSilently(iframeUri, this.baseUrl);
+      const fetchTokenUri = `${document.location.origin}?code=${code}&state=${this.state}`;
```

Comment:
Because state can have arbitrarily value it would be safer to encode it by using `URLSearchParams` or `encodeURIComponent`. Otherwise if it has `=`,`&` or `#` it will brake the url.

Reply:
Oh wow great suggestion, I double checked manually and it was indeed breaking 🙏 This commit should take care of it: 879418e491e76666b836958526344f737b8f3937